core/src/worker/workflow/managed_run.rs

```diff
@@ -71,11 +73,13 @@ pub(super) struct ManagedRun {
     wft: Option<OutstandingTask>,
     /// An outstanding activation to lang
     activation: Option<OutstandingActivation>,
-    /// If set, it indicates there is a buffered poll response from the server that applies to this
-    /// run. This can happen when lang takes too long to complete a task and the task times out, for
-    /// example. Upon next completion, the buffered response will be removed and can be made ready
-    /// to be returned from polling
-    buffered_resp: Option<PermittedWFT>,
+    /// Contains buffered poll responses from the server that apply to this run. This can happen
+    /// when:
+    ///   * Lang takes too long to complete a task and the task times out
+    ///   * Many queries are submitted concurrently and reach this worker (in this case, multiple
+    ///     tasks can be outstanding)
+    ///   * Multiple speculative tasks (ex: for updates) may also exist at once
```

Comment:
Might be worth clarifying this statement that if multiple are sent only the latest is valid

Reply:
Will add